lib/rubocop/cop/rspec/capybara/current_path_expectation.rb

```diff
@@ -30,11 +30,57 @@ class CurrentPathExpectation < Cop
             (send nil? :expect (send {(send nil? :page) nil?} :current_path))
           PATTERN
 
+          # Supported matchers: eq(...) / match(/regexp/) / match('regexp')
+          def_node_matcher :as_is_matcher, <<-PATTERN
+            (send
+              (send nil? :expect (send {(send nil? :page) nil?} :current_path)) ${:to :not_to :to_not}
+              ${(send nil? :eq ...) (send nil? :match (regexp ...))})
+          PATTERN
+
+          def_node_matcher :regexp_str_matcher, <<-PATTERN
+            (send
+              (send nil? :expect (send {(send nil? :page) nil?} :current_path)) ${:to :not_to :to_not}
+              $(send nil? :match (str $_)))
+          PATTERN
+
           def on_send(node)
             expectation_set_on_current_path(node) do
               add_offense(node, location: :selector)
             end
           end
+
+          def autocorrect(node)
+            lambda do |corrector|
+              return unless node.chained?
+              as_is_matcher(node.parent) do |to_sym, matcher_node|
+                rewrite_expectation(corrector, node, to_sym, matcher_node)
+              end
+              regexp_str_matcher(node.parent) do |to_sym, matcher_node, regexp|
```

Comment:
maybe you can combine the 2 matchers in one and just check if the method_name of the matcher_node is `match` and argument is of regexp_type

Reply:
Firstly they was combined, but it was split according to below comment:
https://github.com/rubocop-hq/rubocop-rspec/pull/641#pullrequestreview-129310147

Reply:
I guess my comment “that would reduce duplication in the matchers, but perhaps introduce extra duplication further down in the class…” turned out to be true :-)